src/plugins/data/common/query/types.ts

```diff
@@ -21,10 +21,12 @@ export interface SavedQuery {
   attributes: SavedQueryAttributes;
 }
 
-export interface SavedQueryAttributes {
+// type required for index signature
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type SavedQueryAttributes = {
```

Comment:
This is one of the side-effect of having `T extends SavedObjectAttributes`. We're forced to use TS types instead of interfaces when overloading the default type of `T` to avoid an index signature error. 

```ts
interface SavedObjectAttributes {
  [key: string]: SavedObjectAttribute;
}

interface SavedObject<T extends SavedObjectAttributes = SavedObjectAttributes> { ... }

// valid
let t: SavedObject<{ foo: string; bar: number }>

// valid but eslint error because we should be using an interface
type Foo = { foo: string; bar: number };
let t: SavedObject<Foo>;

// TS2344: Type 'Foo' does not satisfy the constraint 'SavedObjectAttributes'. Index signature for type 'string' is missing in type 'Foo'.
interface Foo { foo: string; bar: number };
let t: SavedObject<Foo>;
```

See https://github.com/microsoft/TypeScript/issues/15300 for more details

it's not really blocker for the changes, but with our lint rules enforcing that we're using interfaces instead of types, this is awkward as it forces to add a linter exception for all type-based SO properties structures

Reply:
we hit a similar issue with persistable state interfaces and went on changing a bunch of interfaces to types